test/e2e/mcp/main_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the License);
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an AS IS BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integ
+// +build integ
+
+package mcp
```

Comment:
@jewertow I noticed with the last commit - shouldn't we use `_test` packages for test code?

Reply:
It was not intended for this PR, it's already quite broad :) 

As with a lot of things, it depends, even [Google's Go Guidelines](https://google.github.io/styleguide/go/decisions#test-package) are a bit vague on that topic... 

I find `package pkg_test` idiom useful because, as an external package, it avoids coupling tests to private details and focuses on the public API. This approach makes refactoring easier, as tests center around top-level features rather than internal implementations. Internal functions can still be tested by deliberately [exporting](https://go.dev/src/runtime/export_test.go) them if needed.

Don't get me wrong - I do not want to enforce it. Just bring it up to consider.